SMLHelper/Handlers/CraftDataHandler_Subnautica.cs

```diff
@@ -206,12 +206,11 @@ void ICraftDataHandler.AddBuildable(TechType techType)
         /// <returns>The ITechData from the modded item if it exists; Otherwise, returns <c>null</c>.</returns>
         ITechData ICraftDataHandler.GetModdedTechData(TechType techType)
         {
-            if (CraftDataPatcher.CustomTechData.TryGetValue(techType, out ITechData moddedTechData))
+            if (!CraftDataPatcher.CustomTechData.TryGetValue(techType, out ITechData moddedTechData))
             {
-                return moddedTechData;
+                moddedTechData = CraftData.Get(techType, true);
             }
-
-            return null;
+            return moddedTechData;
```

Comment:
While I can understand why this could be useful,  
this change effectively is means that `GetModdedTechData` is now allowed to return `TechData` instances that didn't come from mods.  

Rather than change how `ICraftDataHandler.GetModdedTechData` works,  
I would recommend instead creating a new method that does this.

```CS
ITechData GetTechData(TechType techType)
```

Reply:
Ahh yes this was a change when I was trying to make it put the mod data into the games cache and then wipe the custom so that I could just tell it to patch when there was data in the custom data dictionary. Must have forgot to revert that when I changed from that thought